### DISCUSSION TITLE
Expose Metrics Endpoint on Different Port

### DISCUSSION THREAD

**Discussion by vbrandl** [tokio-rs/axum#1443] [Q&A] [answered]
Is there a way to start a second server listening on another port, so a `/metrics` endpoint is not publicly exposed?

The [example](https://github.com/tokio-rs/axum/tree/main/examples/prometheus-metrics) would make the endpoint publicly available.

**Comment by davidpdrsn** [comment, marked as answer]
Use `tokio::spawn` or `join` to run a second server.

**Reply by vbrandl** [reply]
Yeah that works. Thank you.

Should the examples warn that it is bad practice to publicly expose the metrics endpoint or maybe even show a solution for running the metrics endpoint on another port?

If so, I'd be open to create a PR.

**Reply by davidpdrsn** [reply]
Sure!

**Reply by vbrandl** [reply]
I created https://github.com/tokio-rs/axum/pull/1444
